normandy/selfrepair/static/js/self_repair_runner.js

```diff
@@ -27,18 +26,20 @@ function loadAction(recipe) {
     return new Promise((resolve, reject) => {
         let action_name = recipe.action_name;
         if (!registeredActions[action_name]) {
-            xhr.get(`/api/v1/action/${action_name}/`).then(function(response) {
-                let action = JSON.parse(response.responseText);
-                let script = document.createElement('script');
-                script.src = action.implementation_url;
-                script.onload = () => {
-                    if (!registeredActions[action.name]) {
-                        reject(new Error(`Could not find action with name ${action.name}.`));
-                    } else {
-                        resolve(registeredActions[action.name]);
-                    }
-                };
-                document.head.appendChild(script);
+            fetch(`/api/v1/action/${action_name}/`).then(function(response) {
+                return response.json().then(action => {
+                    let script = document.createElement('script');
```

Comment:
I usually do this like

```js
fetch(`/api/v1/action/${action_name}/`)
.then(response => response.json())
.then(action => {
  let script = ...
});
```

Returning `response.json()` directly lets you avoid indenting more.